providers/account.rb

```diff
@@ -234,3 +239,23 @@ def group_resource(exec_action)
       new_resource.updated_by_last_action(true) if r.updated_by_last_action?
   end
 end
+
+def keypair_resource(exec_action)
+  new_resource.ssh_keypair.each do |name, key|
+    # avoid variable scoping issues in resource block
+    key_name, key_content = name, key
+
+    home = Etc.getpwnam(new_resource.username).dir
+    r = file "#{home}/.ssh/#{name}" do
```

Comment:
Ok, push incoming. There's other occasions of paths constructed using string interpolation, though. For example https://github.com/chr4-cookbooks/user/blob/e4959685d6e5db96939c0fb0e6f51bc21901437d/providers/account.rb#L182

Should those be adjusted as well then?